Adds support for OAuth 2.0

samuelgrant opened the pull request:
Adds support for [Oauth 2.0](https://docs.esi.evetech.net/docs/sso/web_based_sso_flow.html). To use v2 set  AuthVersion.v2 in the configuration.

_See: https://github.com/seraphx2/ESI.NET/issues/23#issuecomment-495077911_
```CS
IOptions<EsiConfig> config = Options.Create(new EsiConfig()
{
    EsiUrl = "https://esi.evetech.net/",
    DataSource = DataSource.Tranquility,
    ClientId = "**********",
    SecretKey = "**********",
    CallbackUrl = "",
    UserAgent = "",
    AuthVersion = AuthVersion.v2
});

EsiClient client = new EsiClient(config);```

samuelgrant commented:
I thought I had added JWT Validation but it looks like I missed it. I'll try to add it to my branch tomorrow.

seraphx2 commented:
I am going to be doing some work on ESI finally this weekend. Can you confirm wither this code is complete and ready to be used? Does it conflict with #24 at all?

samuelgrant commented:
The Short Version: NO

I was using your application for API calls but not authentication as I had written my own solution for auth. However, I found I needed to use elements of your SSO code for authenticated ESI calls. I ended up changing your library so I could use CCPs `V2` oauth, but when I opened this PR I had forgotten to include JWT validation. Since then my out of Eve life has exploded and I have stopped playing. 

You are more than welcome to further develop on this branch if my changes are useful, but you will need to implement JWT validation. Please accept my apologies for failing to finish this pull request.


**Done**
- Added an Enum for auth version called `authVersion` which has the values `v1` and `v2`
-  Added the version to the Config class
-  Updated the SSO URL Helper so that it uses v1 auth unless you explicitly configure it to use v2.
-  Updated the Token Helper so it uses v1 auth unless y9ou explicitly configure it to use v2.

**Not Done**
- Configure your Verify method to handle v2 auth including verifying and decrypting JWT tokens.

[Eve SSO V2 docs](https://docs.esi.evetech.net/docs/sso/web_based_sso_flow.html)
